velox/exec/tests/TaskTest.cpp

```diff
@@ -1621,6 +1621,114 @@ DEBUG_ONLY_TEST_F(TaskTest, resumeAfterTaskFinish) {
   waitForAllTasksToBeDeleted();
 }
 
+TEST_F(TaskTest, singleThreadedLongRunningOperatorInTaskReclaimerAbort) {
+  exec::Operator::registerOperator(
+      std::make_unique<TestLongRunningTranslator>());
+  auto data = makeRowVector({
+      makeFlatVector<int64_t>(1'000, [](auto row) { return row; }),
+  });
+
+  // Filter + Project.
+  auto plan =
+      PlanBuilder().values({data, data, data}).project({"c0"}).planFragment();
+
+  auto queryPool = memory::memoryManager()->addRootPool(
+      "blocking-query", 1UL << 30, exec::MemoryReclaimer::create());
+
+  auto queryCtx = std::make_shared<core::QueryCtx>(
+      driverExecutor_.get(),
+      core::QueryConfig{{}},
+      std::unordered_map<std::string, std::shared_ptr<Config>>{},
+      nullptr,
+      std::move(queryPool),
+      nullptr);
+
+  auto blockingTask = Task::create("blocking.task.0", plan, 0, queryCtx);
+
+  // Before we block, we expect `next` to get data normally.
+  EXPECT_NE(nullptr, blockingTask->next());
+
+  // Now, we want to block the pipeline by blocking Values operator. We expect
+  // `next` to return null.  The `future` should be updated for the caller to
+  // wait before calling next() again
+  folly::EventCount getOutputWait;
+  auto getOutputWaitKey = getOutputWait.prepareWait();
```

Comment:
use atomic_bool flag instead of key for event count, and see example in SharedArbitratorTest.cpp